llvm/lib/MC/DXContainerRootSignature.cpp

```diff
@@ -7,17 +7,73 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/MC/DXContainerRootSignature.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/BinaryFormat/DXContainer.h"
 #include "llvm/Support/EndianStream.h"
 
 using namespace llvm;
 using namespace llvm::mcdxbc;
 
+static uint32_t writePlaceholder(raw_svector_ostream &Stream) {
+  const uint32_t DummyValue = std::numeric_limits<uint32_t>::max();
+  uint32_t Offset = Stream.tell();
+  support::endian::write(Stream, DummyValue, llvm::endianness::little);
+  return Offset;
+}
+
+static void rewriteOffset(raw_svector_ostream &Stream, uint32_t Offset) {
+  uint32_t Value =
+      support::endian::byte_swap<uint32_t, llvm::endianness::little>(
+          Stream.tell());
+  Stream.pwrite(reinterpret_cast<const char *>(&Value), sizeof(Value), Offset);
+}
```

Comment:
The name of this function doesn't feel like it conveys exactly what this does. I would probably either make the signature `void rewriteOffset(raw_svector_ostream &Stream, uint32_t Offset, uint32_t Value)` and do the `Stream.tell()` at the caller, or rename this to something like `rewriteOffsetToCurrentByte` that conveys what we're going to write at that location.